### PR TITLE
notification: Drop private GNOME Shell API

### PR DESCRIPTION
**Pull request by jsparber**
Since we have the GNOME specific xdp backend we can move the private
GNOME Shell things there and leave this backend implementation more generic.

### CODE REVIEW

File: src/notification.c [hunk 1]
```diff
@@ -157,6 +157,17 @@ activate_action (GDBusConnection *connection,
                               G_DBUS_CALL_FLAGS_NONE,
                               -1, NULL, NULL, NULL);
 
+      /* The application may not implement the `org.freedesktop.Application`, so
+       * also add the platform data containing the activation-token
+       * to the `ActionInvoked` signal */
```
Comment (TingPing) [inline src/notification.c:104]: This should be documented in `xdg-desktop-portal`?

**Review by TingPing — APPROVED** [review]
LGTM

File: src/notification.c (hunk 1 above)
Comment (jsparber) [reply src/notification.c:104]: Yes, eventually. But I'm not sure about this. I wonder if we should deprecate the `ActionInvoked` signal instead and have apps implement D-Bus Activation

Comment (jsparber) [reply src/notification.c]: I removed it. It's not as important as the other changes and i think need more discussion.

Comment (TingPing) [reply src/notification.c]: It looks to be pretty widely relied upon, from individual apps, to web engines like Chromium and WebKit.

Comment (jsparber) [reply src/notification.c]: > It looks to be relied upon from individual apps to web engines like Chromium and WebKit.

But they will have to change anyways to support the activation-token for wayland activation. So i don't think it's to much to ask to switch to D-Bus activation. Anyhow this discussion doesn't matter for this MR.

Anyhow I forgot that i actually documented the change in the notification v2 MR https://github.com/flatpak/xdg-desktop-portal/pull/1298/commits/440d688de68a7329014a8e0597898ca4f20bb3d1

**Review by GeorgesStavracas — APPROVED** [review]

**Pull request merged**